### PR TITLE
Feature/eco 1017 shipping address in risk check

### PR DESCRIPTION
**Pull request by ievgenvarava**
* Added transfer objects and quote mappers to add DeliveryCustomer into RiskCheck request
* Modified request converter to send new data in RiskCheck request
* Corrected mapers to get salutation from billing/shipping addresses
* Updated tests
* Added new testcase for DeliveryCustomerMapper

### CODE REVIEW

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/Aspect/BillingCustomerMapper.php [hunk 1]
```diff
@@ -44,7 +42,7 @@ public function map(BillingCustomerMapperTransfer $billingCustomerMapperTransfer
         $billingCustomerTransfer->setAddress($address);
         $billingCustomerTransfer->setFirstName($billingAddress->getFirstName());
         $billingCustomerTransfer->setLastName($billingAddress->getLastName());
-        $billingCustomerTransfer->setSalutation(strtoupper($billingCustomerMapperTransfer->getSalutation()));
+        $billingCustomerTransfer->setSalutation(strtoupper($billingAddress->getSalutation()));
```
Comment (profuel) [inline src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/Aspect/BillingCustomerMapper.php]: Minor: please rename billingAddress -> billingAddressTransfer

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/Aspect/DeliveryCustomerMapper.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    /**
+     * @param DeliveryCustomerMapperTransfer $deliveryCustomerMapperTransfer
+     *
+     * @return ArvatoRssDeliveryCustomerTransfer
+     */
+    public function map(DeliveryCustomerMapperTransfer $deliveryCustomerMapperTransfer)
+    {
+        $deliveryCustomerTransfer = new ArvatoRssDeliveryCustomerTransfer();
+        $deliveryAddress = $deliveryCustomerMapperTransfer->getDeliveryAddress();
+        $address = $this->prepareAddressTransfer($deliveryAddress);
```
Comment (profuel) [inline src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/Aspect/DeliveryCustomerMapper.php]: Please call it `$addressTransfer`

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/Aspect/DeliveryCustomerMapper.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * @param DeliveryCustomerMapperTransfer $deliveryCustomerMapperTransfer
+     *
+     * @return ArvatoRssDeliveryCustomerTransfer
+     */
+    public function map(DeliveryCustomerMapperTransfer $deliveryCustomerMapperTransfer)
+    {
+        $deliveryCustomerTransfer = new ArvatoRssDeliveryCustomerTransfer();
+        $deliveryAddress = $deliveryCustomerMapperTransfer->getDeliveryAddress();
```
Comment (profuel) [inline src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/Aspect/DeliveryCustomerMapper.php]: $deliveryAddressTransfer

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/Aspect/DeliveryCustomerMapper.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\AddressTransfer $billingAddress
+     *
+     * @return \Generated\Shared\Transfer\ArvatoRssCustomerAddressTransfer
+     */
+    protected function prepareAddressTransfer(AddressTransfer $billingAddress)
+    {
+        $address = new ArvatoRssCustomerAddressTransfer();
```
Comment (profuel) [inline src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/Aspect/DeliveryCustomerMapper.php]: $addressTransfer

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/RiskCheckRequestMapper.php [hunk 5]
```diff
@@ -29,23 +31,31 @@ class RiskCheckRequestMapper implements RiskCheckRequestMapperInterface
...
     /**
      * @var \SprykerEco\Zed\ArvatoRss\Business\Api\Mapper\Aspect\OrderMapperInterface
      */
     protected $orderMapper;
 
     /**
-     * @param \SprykerEco\Zed\ArvatoRss\Business\Api\Mapper\Aspect\IdentificationMapperInterface $identificationMapper
-     * @param \SprykerEco\Zed\ArvatoRss\Business\Api\Mapper\Aspect\BillingCustomerMapperInterface $billingCustomerMapper
-     * @param \SprykerEco\Zed\ArvatoRss\Business\Api\Mapper\Aspect\OrderMapperInterface $orderMapper
+     * @param IdentificationMapperInterface $identificationMapper
```
Comment (profuel) [inline src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/RiskCheckRequestMapper.php]: Please run sniffer in forced mode to fix class names - should be fully qualified

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/RiskCheckRequestMapper.php [hunk 6]
```diff
@@ -62,12 +72,16 @@ public function mapQuoteToRequestTranfer(QuoteTransfer $quoteTransfer)
         $billingCustomer = $this->billingCustomerMapper->map(
             $this->createBillingCustomerMapperTransfer($quoteTransfer)
         );
+        $deliveryCustomer = $this->deliveryCustomerMapper->map(
```
Comment (profuel) [inline src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/RiskCheckRequestMapper.php]: Same here with name - don't miss `Transfer`

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Converter/RiskCheckRequestConverter.php [hunk 7]
```diff
@@ -51,6 +52,32 @@ protected function convertBillingCustomer(ArvatoRssRiskCheckRequestTransfer $arv
...
+     * @param \Generated\Shared\Transfer\ArvatoRssRiskCheckRequestTransfer $arvatoRssRiskCheckRequestTransfer
+     *
+     * @return array
+     */
+    protected function convertDeliveryCustomer(ArvatoRssRiskCheckRequestTransfer $arvatoRssRiskCheckRequestTransfer)
+    {
+        $result = [];
+        $deliveryCustomerTransfer = $arvatoRssRiskCheckRequestTransfer->getDeliveryCustomer();
+        $addressTranfer = $deliveryCustomerTransfer->getAddress();
+        $address = [
```
Comment (profuel) [inline src/SprykerEco/Zed/ArvatoRss/Business/Api/Converter/RiskCheckRequestConverter.php]: Please extract this to a separate function

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/Aspect/DeliveryCustomerMapper.php [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+        return $deliveryCustomerTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\AddressTransfer $billingAddress
+     *
+     * @return \Generated\Shared\Transfer\ArvatoRssCustomerAddressTransfer
+     */
+    protected function prepareAddressTransfer(AddressTransfer $billingAddress)
```
Comment (profuel) [inline src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/Aspect/DeliveryCustomerMapper.php]: `$billingAddressTransfer`

File: src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/RiskCheckRequestMapper.php (hunk 5 above)
Comment (dereuromark) [reply src/SprykerEco/Zed/ArvatoRss/Business/Api/Mapper/RiskCheckRequestMapper.php]: It's "fix" mode :)

**Review by alex-galych — APPROVED** [review]

**Review by profuel — APPROVED** [review]

**Pull request merged**